mongodbforms/fields.py

```diff
@@ -398,3 +398,40 @@ def _has_changed(self, initial, data):
             if self.contained_field._has_changed(init_val, v):
                 return True
         return False
+
+
+class ListOfFilesField(ListField):
+    widget = ListOfFilesWidget
+
+    def clean(self, value):
+        clean_data = []
+        errors = ErrorList()
+        if not value or isinstance(value, (list, tuple)):
+            if not value or not [
+                    v for v in value if v not in self.empty_values
+            ]:
+                if self.required:
+                    raise ValidationError(self.error_messages['required'])
+                else:
+                    return []
+            if not isinstance(value[0], (list, tuple)):
+                raise ValidationError(self.error_messages['invalid'])
+        else:
+            raise ValidationError(self.error_messages['invalid'])
```

Comment:
Can this be simplified to reduce indent levels with something like

    if not value and not self.required:
        return []

    if value and not isinstance(value, (list, tuple)):
        raise ValidationError(self.error_messages['invalid'])

etc. Simply making sure that the approach is to return a value as quickly as possible by eliminating cases rather than drilling down exhaustively.